admin-dev/themes/new-theme/js/components/form/form-object-mapper.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+const {$} = window;
+
+/**
+ * This is able to watch an HTML form and parse it as a Javascript object based on a configurable
+ * mapping. Each field from the model is mapped to a form input, or several, each input is watched
+ * to keep the model consistent.
+ *
+ * When a model field has several inputs associated it keeps the in sync when either of them is updated.
+ */
+export default class FormObjectMapper {
+  /**
+   * @param $form {jQuery}
+   * @param modelMapping {Object}
+   * @param eventEmitter {EventEmitter}
+   * @param config {Object}
+   */
+  constructor($form, modelMapping, eventEmitter, config) {
+    this.$form = $form;
+    this.fullModelMapping = modelMapping;
+    this.eventEmitter = eventEmitter;
+
+    const inputConfig = config || {};
+
+    // This event is registered so when it is triggered it forces the object update
+    this.updateModelEventName = inputConfig.updateModel || 'updateModel';
+    // This event is emitted each time the object is updated (from both input change and external event)
+    this.modelUpdatedEventName = inputConfig.modelUpdated || 'modelUpdated';
+
+    this.modelMapping = {};
+    this.formMapping = {};
+    this.object = {};
+
+    this.initFormMapping();
+    this.watchUpdates();
+    this.updateFullObject();
+  }
+
+  getObject() {
+    return this.object;
+  }
+
+  /**
+   * Watches if changes happens from the form or via an event.
+   */
+  watchUpdates() {
+    this.$form.on('keyup change dp.change', ':input', (event) => this.inputUpdated(event));
```

Comment:
@jolelievre 

maybe it would be better to use some kind of `debounce` here, just to prevent triggering an event every time we type something/change very quick?

cc @NeOMakinG

Reply:
I agree, and the project has lodash as a dependency, feel free to use _.debounce